enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2506,10 +2534,24 @@ func runCredentialHelper() error {
 	}
 }
 
+// getBazelCommand returns a bazel command (executable and args) with our custom
+// bazelrc options applied.
+func getBazelCommand(rootPath, bazelCmd string, args ...string) ([]string, error) {
+	startupArgs, err := customBazelrcOptions(rootPath, "")
+	if err != nil {
+		return nil, err
+	}
+	args = append([]string{bazelCmd}, append(startupArgs, args...)...)
```

Comment:
It's gross that syscall.Exec requires you to pass bazelCmd as both the executable in the first argument and in the list of arguments. Can you clarify this behavior with a comment, or ideally by default not include bazelCmd in the arguments? In most cases people will probably use exec.CommandContext (which we use in `runCommand`) where we don't need this behavior, and people will likely forget to strip it out of the arguments

Reply:
Also to simplify, we probably don't need to apply the custom .bazelrc for the workspace lock check

Reply:
Changed to not include the bazelCmd in the arguments.

I think we do need the custom .bazelrc since it includes the startup options and I think we need the startup options to match when we run this `--noblock_for_lock` command